cpp/include/rmm/detail/cccl_adaptors.hpp

```diff
@@ -7,21 +7,129 @@
 #include <rmm/cuda_stream_view.hpp>
 #include <rmm/detail/cuda_memory_resource.hpp>
 #include <rmm/detail/export.hpp>
+#include <rmm/mr/detail/device_memory_resource_view.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
```

Comment:
nit: add missing header inclusions for STL utils
```suggestion
#include <rmm/mr/device_memory_resource.hpp>

#include <cstddef>
#include <type_traits>
#include <utility>
```